app/controllers/editorial/nodes_controller.rb

```diff
@@ -18,6 +18,8 @@ def index
 
     def show
       @node = Node.find(params[:id])
+      @section = @node.section
+      @news = NewsArticle.by_section(@section).by_release_date.by_published_at.published.limit(3)
```

Comment:
For testing, maybe you could make this a method/scope on NewsArticle and unit test that?